src/machine.rs

```diff
@@ -695,8 +695,24 @@ impl<'mir, 'tcx> Machine<'mir, 'tcx> for Evaluator<'mir, 'tcx> {
         ptr: Pointer<Self::PointerTag>,
     ) -> InterpResult<'tcx> {
         match ptr.provenance {
-            Tag::Concrete(concrete) =>
-                intptrcast::GlobalStateInner::expose_addr(ecx, concrete.alloc_id),
+            Tag::Concrete(ConcreteTag { alloc_id, sb }) => {
+                intptrcast::GlobalStateInner::expose_addr(ecx, alloc_id);
+
+                let (size, _) =
+                    ecx.get_alloc_size_and_align(alloc_id, AllocCheck::MaybeDead).unwrap();
+
+                // Function pointers and dead objects don't have an alloc_extra so we ignore them.
+                if let Ok(alloc_extra) = ecx.get_alloc_extra(alloc_id) {
+                    if let Some(stacked_borrows) = &alloc_extra.stacked_borrows {
+                        stacked_borrows.ptr_exposed(
```

Comment:
`expose_tag` would be a better name for this function IMO.

Also the function doesn't need to take an `alloc_range` I think.